zerver/lib/webhooks/common.py

```diff
@@ -114,3 +130,104 @@ def validate_extract_webhook_http_header(request: HttpRequest, header: str,
         raise MissingHTTPEventHeader(header)
 
     return extracted_header
+
+class ThirdPartyAPIAmbassador:
+    """ Using an instance of this class, integrations/webhook view code can systematically
+    authenticate with a third party API and then send multiple requests to it, store the results,
+    as well as set triggers to be run before and after the calls. The http_api_callback method of
+    this class is the method that integrations/webhook view code should use to actually call the
+    third party api endpoint. """
+
+    def __init__(self,
+                 bot: UserProfile,
+                 request_kwargs: Dict[str, Any]={},
+                 authentication_handler: Optional[Callable[["ThirdPartyAPIAmbassador"], None]]=None,
+                 request_preprocessor: Optional[Callable[["ThirdPartyAPIAmbassador"], None]]=None,
+                 request_postprocessor: Optional[Callable[["ThirdPartyAPIAmbassador"], None]]=None,
+                 ) -> None:
```

Comment:
This formatting style isn't very readable. Please wrap your arguments on the next line, like so:
``` python
def __init__(
        self, bot: UserProfile,
        request_kwargs: Dict[str, Any]={}
        # and so on
) -> None:

Reply:
Ok, but in my defense I find the style that I generally use to be more readable and I've seen many other places in the codebase use the same exact style, e.g. in this same file:
```python3
def validate_extract_webhook_http_header(request: HttpRequest, header: str,
                                         integration_name: str,
                                         fatal: Optional[bool]=True) -> Optional[str]:
```

But I've made the changes :slightly_smiling_face: